flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/data/TestFlinkParquetWriter.java

```diff
@@ -86,4 +117,141 @@ protected void writeAndValidate(Schema schema) throws IOException {
         RandomGenericData.generateFallbackRecords(schema, NUM_RECORDS, 21124, NUM_RECORDS / 20)),
         schema);
   }
+
+  @Test
+  public void testParquetProperties() throws Exception {
```

Comment:
Instead of using an integration test and lots of custom code to test this, what about using mocks instead to ensure that the Parquet writers are configured using `forTable` or by passing the table's properties?

I think that would be a better test because it separates concerns. The `Parquet` builders should be responsible for translating table settings into correct file config, while the Flink or Spark integration is responsible for configuring the Parquet builder correctly for a given table. The latter is what needs to be tested by Flink.